### PR TITLE
Member milestone

### LINKED ISSUE #119: New online user record message
Send a message to some channel when we achieve a new "online users" record

### PR DESCRIPTION
**Pull request by Armintron**
Emits a message when 100 new members have joined since the last message. Fixes #119

### CODE REVIEW

**Review by tanndlin — DISMISSED** [review]

File: src/app/handlers/membercount.handler.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    const { serverInfo: serverInfoCollection } = collections;
+    const knightEmoji = member.guild.emojis.cache.find((e) => e.name === 'knight');
+
+    if (!serverInfoCollection) {
+      return;
+    }
+
+    const res = await serverInfoCollection
+      .find<IServerCount>({ name: 'MemberCount' })
+      .toArray();
```
Comment (c650) [inline src/app/handlers/membercount.handler.ts:20]: why are you keeping an array?

Comment (tanndlin) [reply src/app/handlers/membercount.handler.ts:20]: Why have more than one entry of the record?
I was planning on storing it, so later on we can see the growth overtime, maybe eventually make a graph or something

File: src/app/handlers/membercount.handler.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+    const lastSavedMemberCount = res[res.length - 1].count;
+
+    if (!res.length) {
+      return;
+    }
```
Comment (tanndlin) [inline src/app/handlers/membercount.handler.ts:26]: Should be above the preceding line

**Review by tanndlin — COMMENTED** [review]
It needs to store the initial count, right now it returns if the length is 0
So make it so that once we hit count % 100 == 0
add it just once, if the collection is empty

File: src/app/handlers/membercount.handler.ts [hunk 3]
```diff
@@ -15,17 +15,29 @@ export class MemberCountHandler implements IHandler {
...
-    const lastSavedMemberCount = res[res.length - 1].count;
+    const countToInsert: IServerCount = {
+      name: 'MemberCount',
+      count: currentCount,
+      dateUpdated: new Date(),
+    };
+
+    if (!memberCountDocs.length) {
+      // we want to add the initial count on a nice number
+      if (currentCount % 100 != 0) {
```
Comment (tanndlin) [inline src/app/handlers/membercount.handler.ts:32]: `!==`

**Review by tanndlin — APPROVED** [review]

**Pull request merged**